qa/TL3_JAX_multiprocess/test_jax.sh

```diff
@@ -6,4 +6,18 @@ function CLEAN_AND_EXIT {
 
 python -c "import jax; print(jax.devices()); assert jax.device_count() > 0"
 
+echo "Test one GPU per process"
+CUDA_VISIBLE_DEVICES="1" python jax_client.py --id 1 --size 8 &
+CUDA_VISIBLE_DEVICES="2" python jax_client.py --id 2 --size 8 &
+CUDA_VISIBLE_DEVICES="3" python jax_client.py --id 3 --size 8 &
+CUDA_VISIBLE_DEVICES="4" python jax_client.py --id 4 --size 8 &
+CUDA_VISIBLE_DEVICES="5" python jax_client.py --id 5 --size 8 &
+CUDA_VISIBLE_DEVICES="6" python jax_client.py --id 6 --size 8 &
+CUDA_VISIBLE_DEVICES="7" python jax_client.py --id 7 --size 8 &
```

Comment:
I think that if we are running things in background that may in some pathological cases leads to hangs or false positives in the test job. 
It would probably be good to capture all the started processes and wait for their completion or terminate if this doesn't happen in reasonable time.

Reply:
Main job is not in the background and it will succeed only if all the clients will connect successfully, otherwise it will fail.
I am all for making it more robust so can you elaborate a bit more about the setup that you have in mind what would work better?

Reply:
I was thinking at least to wait for all jobs to correctly terminate like here: https://stackoverflow.com/a/73620420

Reply:
I read a bit and I come up with something that I think is better. I am using timeout to exit the test even when there is some hang in there and I am waiting for all the jobs.